test/ttmlir/Dialect/TTNN/all_l1_interleaved_policy.mlir

```diff
@@ -0,0 +1,46 @@
+// RUN: ttmlir-opt --ttnn-optimizer="memory-layout-analysis-enabled=true memory-layout-analysis-policy=L1Interleaved" %s | FileCheck %s
```

Comment:
We don't need duplicate tests in dialect and silicon. If they are testing the same thing, silicon tests are sufficien.